### PR TITLE
Add fictionpress source, based on fanfiction source

### PR DESCRIPTION
**Pull request by watzeedzad**
- Add fictionpress.com, code based on fanfiction.net

### CODE REVIEW

**Comment by watzeedzad** [comment]
#1112

File: sources/fictionpress.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        logger.info('Novel title: %s', self.novel_title)
+
+        possible_image = soup.select_one('#profile_top img.cimage')
+        if possible_image:
+            self.novel_cover = self.absolute_url(possible_image['src'])
+        # end if
+        logger.info('Novel cover: %s', self.novel_cover)
+
+        self.novel_author = soup.select_one(
+            '#profile_top, #content').select_one('a[href*="/u/"]').text.strip()
```
Comment (dipu-bd) [inline sources/fictionpress.py:53]: you should handle the case where elements are not present. e.g.

```python
possible_author = soup.select_one('#profile_top, #content')
if possible_author:
    possible_author = soup.select_one('a[href*="/u/"]')
if possible_author:
    self.author = possible_author.text.strip()
```

**Comment by watzeedzad** [comment]
Oh I saw you fix it already, Sorry for late reply but I was busy with my work yesterday so I didn't have time to take a look at it.

**Comment by dipu-bd** [comment]
No problem. Thanks for contributing.

**Pull request merged**